microcosm_flask/conventions/config.py

```diff
@@ -0,0 +1,83 @@
+"""
+Config sharing convention.
+
+Reports service config protected by basic auth for securely running services
+locally with realistic config.
+
+"""
+from distutils.util import strtobool
+from microcosm.api import defaults
+from json import dumps, loads
+
+from microcosm_flask.audit import skip_logging
+from microcosm_flask.conventions.base import Convention
+from microcosm_flask.conventions.encoding import make_response
+from microcosm_flask.namespaces import Namespace
+from microcosm_flask.operations import Operation
+
+
+class Config:
+    """
+    Wrapper around service config state.
+
+    """
+    def __init__(self, graph, include_build_info=True):
+        self.graph = graph
+        self.name = graph.metadata.name
+        self.config = dict(
+            self.graph.config
+        )
+
+    def to_dict(self):
```

Comment:
This is the most questionable part. We could probably inject a custom json encoder in a different part of the stack, but not sure we even like this direction in general so just putting this idea out there with something simple.

Reply:
I think we need to do some work in the config layer to make this safe (e.g. labeling secrets). We have a few other ideas around config improvements (e.g. validation of required values and auto-typing). We're probably do for a design session there.